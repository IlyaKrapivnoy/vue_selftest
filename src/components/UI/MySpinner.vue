<template>
  <div class="flex items-center justify-center">
    <div
      class="animate-spin rounded-full h-6 w-6 border-t-4 border-blue-500"
    ></div>
  </div>
</template>
<script>
export default {
  name: "my-spinner",
};
</script>
