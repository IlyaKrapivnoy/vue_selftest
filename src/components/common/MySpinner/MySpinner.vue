<template>
  <div :class="['flex items-center justify-center my-10', wrapperClass]">
    <div class="animate-spin rounded-full h-6 w-6 border-t-4 border-blue-500" />
  </div>
</template>

<script>
export default {
  name: "MySpinner",
  props: {
    wrapperClass: {
      type: String,
      default: "",
    },
  },
};
</script>
