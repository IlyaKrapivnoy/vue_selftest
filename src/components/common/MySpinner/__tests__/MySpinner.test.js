import { mount } from "@vue/test-utils";
import MySpinner from "@/components/common/MySpinner/MySpinner.vue";

describe("MySpinner", () => {
  it("matches snapshot", () => {
    const wrapper = mount(MySpinner);
    expect(wrapper.html()).toMatchSnapshot();
  });

  it("renders the spinner element", () => {
    const wrapper = mount(MySpinner);
    const spinnerElement = wrapper.find(".animate-spin");
    expect(spinnerElement.exists()).toBe(true);
  });

  it("applies the provided CSS class to the wrapper div", () => {
    const customClass = "my-custom-class";
    const wrapper = mount(MySpinner, {
      props: {
        class: customClass,
      },
    });
    expect(wrapper.classes()).toContain(customClass);
  });

  it("renders with default name", () => {
    const wrapper = mount(MySpinner);
    expect(wrapper.vm.$options.name).toBe("my-spinner");
  });

  it("has a default prop for wrapper class", () => {
    const wrapper = mount(MySpinner);
    expect(wrapper.props("wrapperClass")).toBeUndefined();
  });
});
