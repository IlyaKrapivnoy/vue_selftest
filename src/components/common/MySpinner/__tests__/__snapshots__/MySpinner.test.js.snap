// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MySpinner matches snapshot 1`] = `
"<div class="flex items-center justify-center my-10">
  <div class="animate-spin rounded-full h-6 w-6 border-t-4 border-blue-500"></div>
</div>"
`;
